tests/plenary/path_spec.lua

```diff
@@ -205,6 +205,89 @@ describe('Path', function()
     end)
   end)
 
+  describe('rename', function()
+    it('can rename a file', function()
+      local p = Path:new("a_random_filename.lua")
+      assert(pcall(p.touch, p))
+      assert(p:exists())
+
+      assert(pcall(p.rename, p, { new_name = "not_a_random_filename.lua" }))
+      assert.are.same("not_a_random_filename.lua", p.filename)
+
+      p:rm()
+    end)
+
+    it('can handle an invalid filename', function()
+      local p = Path:new("some_random_filename.lua")
+      assert(pcall(p.touch, p))
+      assert(p:exists())
+
+      assert(not pcall(p.rename, p, { new_name = "" }))
+      assert(not pcall(p.rename, p))
+      assert.are.same("some_random_filename.lua", p.filename)
+
+      p:rm()
+    end)
+
+    it('can move to parent dir', function()
+      local p = Path:new("some_random_filename.lua")
+      assert(pcall(p.touch, p))
+      assert(p:exists())
+
+      assert(pcall(p.rename, p, { new_name = "../some_random_filename.lua" }))
+      assert.are.same("some_random_filename.lua", p.filename)
+
+      p:rm()
```

Comment:
Something didn't clean up here. So i could only ran the test once

Reply:
So yeah its not being cleaned up because the new absolute path is still `/home/conni/repos/plenary.nvim/some_random_filename.lua` rather than `/home/conni/repos/some_random_filename.lua`

Thats the only thing i found, so really good work. :) good job @elianiva 
And of course thank you :)